src/app/files/file-explorer.js

```diff
@@ -138,10 +162,9 @@ function fileExplorer (appAPI, files) {
   var textUnderEdit = null
   var textInRename = false
 
-  var events = new EventManager()
-  this.events = events
-  var api = {}
-  api.addFile = function addFile (file) {
+  self.events = new EventManager()
+  self.api = {}
+  self.api.addFile = function addFile (file) {
```

Comment:
the `addFile`  button seems to be broken

Reply:
hm - i didn't touch the `addFile` button, but it's probably related to emitting an event - and i think i added `@TODO` tags to those - so yes that needs fixing

Reply:
Let's fix in the next PR